planner/core/find_best_task.go

```diff
@@ -624,14 +624,29 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 	}
 
 	if ds.ctx.GetSessionVars().GetAllowPreferRangeScan() && len(candidates) > 1 {
-		// remove the table/index full scan path
-		for i, c := range candidates {
-			for _, ran := range c.path.Ranges {
-				if ran.IsFullRange() {
-					candidates = append(candidates[:i], candidates[i+1:]...)
-					return candidates
+		// For those paths which are neither TiFlash-path nor forced-path, we remove table/index full scan paths if there
+		// exists some range scan path.
```

Comment:
I think this sentence is not very fluent. :sweat_smile:

Reply:
PTAL at the new comment, I'm not sure whether the comment is clear enough.